scipy/sparse/linalg/eigen/_svds.py

```diff
@@ -62,40 +141,60 @@ def svds(A, k=6, ncv=None, tol=0, which='LM', v0=None,
         When ``solver='arpack'``, this is the number of Lanczos vectors
         generated. See :ref:`'arpack' <sparse.linalg.svds-arpack>` for details.
-        When ``solver='lobpcg'``, this parameter is ignored.
+        When ``solver='lobpcg'`` or ``solver='propack'``, this parameter is
+        ignored.
     tol : float, optional
         Tolerance for singular values. Zero (default) means machine precision.
     which : {'LM', 'SM'}
         Which `k` singular values to find: either the largest magnitude ('LM')
         or smallest magnitude ('SM') singular values.
     v0 : ndarray, optional
         The starting vector for iteration; see method-specific
-        documentation (:ref:`'arpack' <sparse.linalg.svds-arpack>` or
-        :ref:`'lobpcg' <sparse.linalg.svds-lobpcg>`) for details.
+        documentation (:ref:`'arpack' <sparse.linalg.svds-arpack>`,
+        :ref:`'lobpcg' <sparse.linalg.svds-lobpcg>`), or
+        :ref:`'propack' <sparse.linalg.svds-propack>` for details.
     maxiter : int, optional
         Maximum number of iterations; see method-specific
-        documentation (:ref:`'arpack' <sparse.linalg.svds-arpack>` or
-        :ref:`'lobpcg' <sparse.linalg.svds-lobpcg>`) for details.
-    return_singular_vectors : bool or str, optional
+        documentation (:ref:`'arpack' <sparse.linalg.svds-arpack>`,
+        :ref:`'lobpcg' <sparse.linalg.svds-lobpcg>`), or
+        :ref:`'propack' <sparse.linalg.svds-propack>` for details.
+    return_singular_vectors : {True, False, "u", "vh"}
         Singular values are always computed and returned; this parameter
         controls the computation and return of singular vectors.
 
         - ``True``: return singular vectors.
         - ``False``: do not return singular vectors.
-        - ``"u"``: only return the left singular values, without computing the
-          right singular vectors (if ``N > M``).
-        - ``"vh"``: only return the right singular values, without computing
-          the left singular vectors (if ``N <= M``).
+        - ``"u"``: if ``M <= N``, compute only the left singular vectors and
+          return ``None`` for the right singular vectors. Otherwise, compute
+          all singular vectors.
+        - ``"vh"``: if ``M > N``, compute only the right singular vectors and
+          return ``None` for the left singular vectors. Otherwise, compute
+          all singular vectors.
+
+        If ``solver='propack'``, the option is respected regardless of the
+        matrix shape.
 
     solver : str, optional
             The solver used.
-            :ref:`'arpack' <sparse.linalg.svds-arpack>` and
-            :ref:`'lobpcg' <sparse.linalg.svds-lobpcg>` are supported.
+            :ref:`'arpack' <sparse.linalg.svds-arpack>`,
+            :ref:`'lobpcg' <sparse.linalg.svds-lobpcg>`, and
+            :ref:`'propack' <sparse.linalg.svds-propack>` are supported.
             Default: `'arpack'`.
+    random_state : {None, int, `numpy.random.Generator`,
```

Comment:
@mdhaber I was read through the discussion in gh-13778 and wonder if we should doc-deprecate the `numpy.random.RandomState` usage here anticipating reviewer comments

Reply:
If that means to just remove it from the documentation, I'm ok with that.

Reply:
Yeah, that's what I meant by doc-deprecate.  There's some discussion going on now in the UNU.RAN PR that I'd like to see play out before we make any changes.  Might not need to do anything

Reply:
K.